packages/gatsby-theme-store/src/sdk/search/controller.ts

```diff
@@ -22,14 +22,58 @@ export const history = {
 
 export type SearchHistory = typeof history
 
-export const search = (term: string) => {
-  history.add(term)
+const pathExists = async (pathname: string) => {
+  try {
+    const response = await fetch(`/page-data${pathname}/page-data.json`, {
+      redirect: 'error',
+    })
+
+    if (response.status !== 200) {
+      return false
+    }
+
+    return true
+  } catch (err) {
+    return false
+  }
+}
 
+/**
+ * https://help.vtex.com/en/tutorial/how-does-vtex-search-work--tutorials_542#vtex-search-engine-product-display-prioritization
+ *
+ * To be compliant with "VTEX search engine product display prioritization" as described in the link above we implement the following
+ * algorithm:
+ *
+ * 1. Make a call to the page's data object.
+ * 2. If it exists, this means we Server Side Generated (SSG) a page for this search term and we should navigate there instead
+ * 3. If it doesn't exist, we should fallback to a full text search instead
+ *
+ * This "VTEX search engine product display prioritization" is important because it allows our customers to create landing pages
+ * for their most researched terms, crafting a custom experience for their best sellers.
+ * A cool side effect of this approach is warming the browser's cache with the page's data object in case of the term having an
+ * specific landing page
+ */
+export const search = async (term: string) => {
+  let pathname = ''
   const params = new URLSearchParams(window.location.search)
 
-  params.set('map', 'ft')
+  // Check if "term" has a more specific page
+  const slugified = encodeURIComponent(term.toLowerCase().replace(/ /g, '-'))
```

Comment:
Should we worry about other types of punctuation and accentuation? Like a search for "açúcar e adoçante"

Reply:
Nice @lariciamota ! Just copied the right slugify function from https://github.com/vtex/rewriter/blob/a7f288cc47c3921e03f44c8eb75de66e95829f8a/node/clients/catalog.ts#L72